### PR TITLE
[144] New freak query node

### PR DESCRIPTION
**Pull request by gabrielgiroe1**
## Description
In this PR we can query for a freak with a specific id

## Related

[[42] [New "freak" query node](https://trello.com/c/tOINYKAA/144-new-freak-query-node)

## Steps to Test or Reproduce
Go in insomnia and query a freak by id
```
query{
  freak(id: 1){
    description
    id
  }
}
```

### CODE REVIEW

**Review by salcudeann — APPROVED** [review]

File: spec/fixtures/responses/queries/freak/freak.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "data": {
+    "freak": {
+      "description": "lalala"
+    }
```
Comment (alex-damian-negru) [inline spec/fixtures/responses/queries/freak/freak.json:5]: Where is this data coming from? I don't see any factory being called anywhere in the test.

Comment (gabrielgiroe1) [reply spec/fixtures/responses/queries/freak/freak.json:5]: I don't know. :))

File: db/Guardfile [hunk 2]
```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
```
Comment (alex-damian-negru) [inline db/Guardfile:1]: Why is this file popping up in all of your PRs? :))

I really don't expect changes to it to happen so often. Please don't forget to do self code reviews.

File: spec/controllers/graphql/queries/freaks_spec.rb [hunk 3]
```diff
@@ -12,12 +12,14 @@ module Graphql
       }
     end
 
-    before do
-      create(:freak, first_name: 'Ion')
-      create(:freak, first_name: 'Gheorghe')
-    end
+    context 'when freak create' do
```
Comment (alex-damian-negru) [inline spec/controllers/graphql/queries/freaks_spec.rb:15]: Much english :))

File: spec/support/matchers/controller_matchers.rb [hunk 4]
```diff
@@ -46,7 +46,10 @@ def ensure_dir(path)
   end
 
   def write_new_fixture(path)
-    File.write(path, serialized_response) if !File.exist?(path) && !File.exist?("#{path}.erb")
+    return if File.exist?(path) || File.exist?("#{path}.erb")
+    return if ENV['DISABLE_FIXTURE_CAPTURE']
```
Comment (alex-damian-negru) [inline spec/support/matchers/controller_matchers.rb:50]: We can combine this on a single line.

File: spec/fixtures/responses/queries/freaks/freaks_pagination.json [hunk 5]
```diff
@@ -4,91 +4,13 @@
     "data": {
       "freaks": {
         "pageInfo": {
-          "endCursor": "MQ",
-          "startCursor": "MQ",
+          "endCursor": null,
+          "startCursor": null,
```
Comment (alex-damian-negru) [inline spec/fixtures/responses/queries/freaks/freaks_pagination.json:8]: This is wrong. Why delete the expected response? This file & it's spec should suffer no changes.

File: spec/fixtures/responses/queries/freaks/freaks.json [hunk 6]
```diff
@@ -17,10 +17,6 @@
               "lastName": "doe",
               "description": "a simple freak",
               "email": "freak@gmail.com",
-              "photo": {
```
Comment (alex-damian-negru) [inline spec/fixtures/responses/queries/freaks/freaks.json:20]: Why delete the `photo`s here?

File: spec/fixtures/requests/queries/freaks.graphql [hunk 7]
```diff
@@ -13,10 +13,6 @@ query Freaks($last: Int, $before: String) {
                 lastName
                 description
                 email
-                photo {
```
Comment (alex-damian-negru) [inline spec/fixtures/requests/queries/freaks.graphql:16]: Same here

**Review by alex-damian-negru — CHANGES_REQUESTED** [review]

File: db/Guardfile (hunk 2 above)
Comment (gabrielgiroe1) [reply db/Guardfile:1]: ok

Comment (alex-damian-negru) [reply db/Guardfile:1]: image

File: spec/controllers/graphql/queries/freak_spec.rb [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    subject(:query_freaks) { post :execute, params: params, as: :json }
+
+    let(:params) do
+      {
+        query: File.read('spec/fixtures/requests/queries/freak.graphql')
+      }
+    end
+
+    before do
+      create(:freak, id: 1, first_name: 'Ion', description: 'the best')
```
Comment (alex-damian-negru) [inline spec/controllers/graphql/queries/freak_spec.rb:16]: Is it necessary to specify the `id` here? 

From experience, it's a bad practice to specify id's when creating entities.

File: spec/fixtures/responses/queries/freaks/freaks.json (hunk 6 above)
Comment (alex-damian-negru) [reply spec/fixtures/responses/queries/freaks/freaks.json:20]: image

**Review by alex-damian-negru — CHANGES_REQUESTED** [review]

**Review by alex-damian-negru — APPROVED** [review]
Looks good.

**Pull request merged**